data/yam/agama/auto/template.libsonnet

```diff
@@ -40,6 +40,7 @@ function(addon_ha_reg_code='',
     [if scripts_post_partitioning != '' then 'postPartitioning']: [ scripts_post_partitioning_lib[x] for x in std.split(scripts_post_partitioning, ',') ],
     [if scripts_pre != '' then 'pre']: [ scripts_pre_lib[x] for x in std.split(scripts_pre, ',') ],
   },
+  [if std.startsWith(storage, 'raid') then 'storage']: storage_lib[storage],
```

Comment:
not sure this select anything.

Reply:
I tried to run a vr with raid1 by adding a line "raid1: raid('raid1')" in storage.libsonnet, the generated profile looks fine, though the VR is not green:
https://openqa.suse.de/tests/17916635#step/boot_agama/2
Then does it mean this line works fine?

Reply:
test data is prepared to accept default values, to reuse the same test data you need to follow:
https://progress.opensuse.org/issues/183002#note-10

Reply:
that this line really insert the raid, and does it only for this case or for any case, for me looks like we are not calling the specific function but I might be wrong.

Reply:
I don't know how this could work, 
`[if std.startsWith(storage, 'raid') then 'storage']: storage_lib[storage],` 
aren't you adding the whole storage file here?

Reply:
It means if the storage variable starts with "raid", then a field named "storage" is added to the object.
The value assigned is `storage_lib[storage]`, which should be a value (likely an object or configuration) corresponding to the specific RAID type from the storage_lib map.
It does not add the whole `storage_lib` file, but just the entry indexed by storage.
If, for example, storage is "`raid1`", it adds "storage": `storage_lib["raid1"] `to the output object.

If `storage_lib[storage]` is an object describing the RAID setup, that object will be added under the "storage" key. If it’s not defined, the field won’t be added.